consent-enforcement/consent-enforcement-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/consent/enforcement/ConsentEnforcementPayloadMediator.java

```diff
@@ -0,0 +1,100 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.financial.services.apim.mediation.policies.consent.enforcement.constants.ConsentEnforcementConstants;
+import org.wso2.financial.services.apim.mediation.policies.consent.enforcement.utils.ConsentEnforcementUtils;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Mediator to generate the payload required to be sent for the consent validation service.
+ */
+public class ConsentEnforcementPayloadMediator extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(ConsentEnforcementPayloadMediator.class);
+
+    private String consentIdClaimName;
+
+    @Override
+    public boolean mediate(MessageContext messageContext) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Map<String, Object> headers = (Map<String, Object>)
+                axis2MessageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+
+        String extractedConsentId;
+        try {
+            extractedConsentId = ConsentEnforcementUtils.extractConsentIdFromJwtToken(headers, consentIdClaimName);
+        } catch (UnsupportedEncodingException e) {
+            // TODO: handle error properly
+            return false;
+        }
+
+        if (StringUtils.isBlank(extractedConsentId)) {
+            // TODO: handle error properly
+            return false;
+        }
+
+        Map<String, Object> additionalParams = new HashMap<>();
+        additionalParams.put(ConsentEnforcementConstants.CONSENT_ID_TAG,
+                extractedConsentId);
+        additionalParams.put(ConsentEnforcementConstants.ELECTED_RESOURCE_TAG,
+                messageContext.getProperty(ConsentEnforcementConstants.API_ELECTED_RESOURCE));
+        additionalParams.put(ConsentEnforcementConstants.RESOURCE_PARAMS_TAG,
+                ConsentEnforcementUtils.getResourceParamMap(messageContext));
+         additionalParams.put(ConsentEnforcementConstants.USER_ID_TAG,
+                 "is_admin@wso2.com"/*messageContext.getProperty(ConsentEnforcementConstants.USER_ID)*/);
+         additionalParams.put(ConsentEnforcementConstants.CLIENT_ID_TAG,
+                 "123"/*messageContext.getProperty(ConsentEnforcementConstants.CONSUMER_KEY)*/);
+
+        JSONObject validationRequest;
+        try {
+            validationRequest = ConsentEnforcementUtils
+                    .createValidationRequestPayload(axis2MessageContext, headers, additionalParams);
```

Comment:
2 things.. 1) why we need to pass headers seperatly.Those can be derived from axis2 message context right? 2) And additional params also can derive from axis2Message Context.. so can we change either passing only axis2MessageContext or remove passing axis2MessageContext and pass specific params..